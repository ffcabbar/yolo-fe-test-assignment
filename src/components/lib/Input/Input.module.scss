.input {
  width: 100%;
  border: 1px solid #a4a4a4;
  border-radius: 0.2rem;
  outline-color: #ff4700;
  padding: 0.8rem;
  font-size: 14px;
  color: black;
  font-weight: 600;

  &::placeholder {
    color: #8492a6;
  }
}
