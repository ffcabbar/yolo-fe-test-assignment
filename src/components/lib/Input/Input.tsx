import styles from './Input.module.scss';

type Props = React.DetailedHTMLProps<React.InputHTMLAttributes<HTMLInputElement>, HTMLInputElement>;

export const Input = ({ ...rest }: Props) => {
  return (
    <div>
      <input role="searchbox" className={styles.input} {...rest} />
    </div>
  );
};
