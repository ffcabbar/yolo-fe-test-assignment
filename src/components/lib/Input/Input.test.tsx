import { fireEvent, render, screen } from '@testing-library/react';
import { useState } from 'react';
import { Input } from './Input';
import renderer from 'react-test-renderer';

function TestInput() {
  const [value, setValue] = useState('');

  const handleChange = (e: any) => {
    setValue(e.target.value);
  };

  return <Input value={value} aria-label="test-input" onChange={handleChange} />;
}

const setup = () => {
  const utils = render(<TestInput />);
  const input = utils.getByLabelText('test-input');
  return {
    input,
    ...utils
  };
};

test('handles onChange', () => {
  const { input } = setup();
  fireEvent.change(input, { target: { value: 'Yolo Group' } });
  expect((input as HTMLInputElement).value).toBe('Yolo Group');
});

test('renders input', () => {
  render(<Input />);
  const inputElement = screen.getByRole('searchbox');
  expect(inputElement).toBeInTheDocument();
});

test('matches snapshot', () => {
  const tree = renderer.create(<Input placeholder="BTC" />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('matches snapshot correctly when there are no props', () => {
  const tree = renderer.create(<Input />).toJSON();
  expect(tree).toMatchSnapshot();
});
