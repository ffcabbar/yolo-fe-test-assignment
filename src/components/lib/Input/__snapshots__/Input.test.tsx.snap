// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <input
    className="input"
    placeholder="BTC"
    role="searchbox"
  />
</div>
`;

exports[`matches snapshot correctly when there are no props 1`] = `
<div>
  <input
    className="input"
    role="searchbox"
  />
</div>
`;
